core/src/test/java/org/elasticsearch/index/query/AndQueryBuilderTest.java

```diff
@@ -116,4 +118,38 @@ public void testValidate() {
         }
         assertValidate(andQuery, totalExpectedErrors);
     }
+
+    public void testParsingToplevelArray() throws IOException {
+        QueryParseContext context = createParseContext();
+        String queryString = "{ \"and\" : [ { \"match_all\" : {} } ] }";
+        XContentParser parser = XContentFactory.xContent(queryString).createParser(queryString);
+        context.reset(parser);
+        assertThat(parser.nextToken(), is(XContentParser.Token.START_OBJECT));
+        assertThat(parser.nextToken(), is(XContentParser.Token.FIELD_NAME));
+        assertThat(parser.currentName(), is(AndQueryBuilder.NAME));
```

Comment:
Because the assertQueryHeader() helper expects an additional START_OBJECT token which in this special case of the query syntax we don't have.